litewallet/BuyTableViewController.swift

```diff
@@ -0,0 +1,151 @@
+		sfSafariVC.delegate = self
+		present(sfSafariVC, animated: true)
+	}
+
+	// MARK: Simplex UI
+
+	@IBOutlet var simplexLogoImageView: UIImageView!
+	@IBOutlet var simplexHeaderLabel: UILabel!
+	@IBOutlet var simplexDetailsLabel: UILabel!
+	@IBOutlet var simplexCellContainerView: UIView!
+	@IBOutlet var simplexCurrencySegmentedControl: UISegmentedControl!
+
+	private var currencyCode: String = "USD"
+	private let uuidString: String = UIDevice.current.identifierForVendor?.uuidString ?? ""
+	private let currentWalletAddress: String = WalletManager.sharedInstance.wallet?.receiveAddress ?? ""
+
+	@IBAction func didTapSimplex(_: Any) {
+		if let vcWKVC = UIStoryboard(name: "Buy", bundle: nil).instantiateViewController(withIdentifier: "BuyWKWebViewController") as? BuyWKWebViewController {
+			vcWKVC.currencyCode = currencyCode
+			vcWKVC.currentWalletAddress = currentWalletAddress
+			vcWKVC.uuidString = uuidString
+			vcWKVC.timestamp = Int(Date().timeIntervalSince1970)
+			addChild(vcWKVC)
+			view.addSubview(vcWKVC.view)
+			vcWKVC.didMove(toParent: self)
+
+			vcWKVC.didDismissChildView = {
+				for vc in self.children {
+					DispatchQueue.main.async {
+						vc.willMove(toParent: nil)
+						vc.view.removeFromSuperview()
+						vc.removeFromParent()
+					}
+				}
+			}
+		} else {
+			NSLog("ERROR: Storyboard not initialized")
+		}
+	}
+
+	var store: Store?
+	var walletManager: WalletManager?
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		let thinHeaderView = UIView()
+		thinHeaderView.frame = CGRect(x: 0, y: 0, width: view.frame.width, height: 1.0)
+		thinHeaderView.backgroundColor = .white
+		tableView.tableHeaderView = thinHeaderView
+		tableView.tableFooterView = UIView()
+
+		moonpaySegmentedControl.addTarget(self, action: #selector(didChangeCurrencyMoonpay), for: .valueChanged)
+		moonpaySegmentedControl.selectedSegmentIndex = PartnerFiatOptions.usd.index
+		moonpaySegmentedControl.selectedSegmentTintColor = .white
+		moonpaySegmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.white], for: .normal)
+		moonpaySegmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.liteWalletBlue], for: .selected)
+
+		simplexCurrencySegmentedControl.addTarget(self, action: #selector(didChangeCurrencySimplex), for: .valueChanged)
+		simplexCurrencySegmentedControl.selectedSegmentIndex = PartnerFiatOptions.usd.index
+		simplexCurrencySegmentedControl.selectedSegmentTintColor = .white
+		simplexCurrencySegmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.white], for: .normal)
+		simplexCurrencySegmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.liteWalletBlue], for: .selected)
+
+		setupWkVCData()
+
+		LWAnalytics.logEventWithParameters(itemName: ._20191105_DTBT)
+	}
+
+	private func setupWkVCData() {
+		let bitrefillData = Partner.partnerDataArray()[0]
```

Comment:
The app might crash if any changes happen to this `partnerDataArray`. I would suggest adding a condition before you're able to access the value inside to make sure it exists.